procsim/biomass/test/test_raw_product_generator.py

```diff
@@ -224,6 +224,84 @@ def testMergeToSingleSlice(self):
         self.assertFalse(start_aligned)
         self.assertFalse(end_aligned)
 
+    def testNoDataTakeInfo(self) -> None:
+        '''Production should fail if no data take info is found.'''
+        gen = self.create_class_under_test()
+        # Remove mandatory data take info.
+        del gen._scenario_config['data_take_id']
+        gen.read_scenario_parameters()
+
+        # Normally we read this from input products, but now we set it by hand.
+        begin = datetime.datetime(2021, 2, 1, 0, 24, 32, 0, tzinfo=datetime.timezone.utc)
+        end = datetime.datetime(2021, 2, 1, 0, 29, 32, 0, tzinfo=datetime.timezone.utc)
+        gen._hdr.validity_start = gen._hdr.begin_position = begin
+        gen._hdr.validity_stop = gen._hdr.end_position = end
+
+        with self.assertRaises(ScenarioError):
+            gen.generate_output()
+
+    def testMultipleDataTakes(self) -> None:
+        '''
+        Test that multiple data takes appear in the MPH and that the
```

Comment:
In the current implementation, only a single datatake_id can appear in a MPH, right?

Reply:
Indeed, this was an outdated docstring! I've updated it.